orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/impl/OrientDBSchemaDefiner.java

```diff
@@ -0,0 +1,547 @@
+
+		Set<Class> set2 = new HashSet<>();
+		set2.add( ManyToOneType.class );
+		set2.add( OneToOneType.class );
+		RELATIONS_TYPES = Collections.unmodifiableSet( set2 );
+
+	}
+
+	private String createClassQuery(String tableName) {
+		return String.format( "create class %s extends V", tableName );
+	}
+
+	private String createClassQuery(Table table) {
+		String query = null;
+		if ( isTablePerClassInheritance( table ) ) {
+			query = String.format( "create class %s extends %s", table.getName(), table.getPrimaryKey().getTable().getName() );
+
+		}
+		else {
+			query = String.format( "create class %s extends V", table.getName() );
+		}
+		return query;
+	}
+
+	private void createSequence(Connection connection, String name, int startValue, int incValue) {
+		try {
+			String query = String.format( "CREATE SEQUENCE %s TYPE ORDERED START %d INCREMENT %d", name, ( startValue == 0 ? 0 : startValue - 1 ), incValue );
+			log.debugf( "query for create sequnce: %s", query );
+			connection.createStatement().execute( query );
+		}
+		catch (OSequenceException ose) {
+			log.warnf( "OSequenceException: %s", ose.getMessage() );
+		}
+		catch (SQLException | OException e) {
+			throw log.cannotGenerateSequence( name, e );
+		}
+	}
+
+	private void createTableSequence(Connection connection, String seqTable, String pkColumnName, String valueColumnName) {
+		try {
+			connection.createStatement().execute( String.format( "create class %s extends V", seqTable ) );
+			connection.createStatement().execute( String.format( "create property %s.%s string ", seqTable, pkColumnName ) );
+			connection.createStatement().execute( String.format( "create property %s.%s long ", seqTable, valueColumnName ) );
+			connection.createStatement().execute( String.format( "create index %s.%s unique ", seqTable, pkColumnName ) );
+		}
+		catch (SQLException | OException e) {
+			throw log.cannotGenerateClass( seqTable, e );
+		}
+	}
+        
+	private void createGetTableSeqValueFunc(Connection connection) {
+		try {
+			OrientJdbcConnection orientDBConnection = (OrientJdbcConnection) connection;                        
+			Set<String> functions = orientDBConnection.getDatabase().getMetadata().getFunctionLibrary().getFunctionNames();
+			log.debugf( " functions : %s", functions );
+			if ( functions.contains( "getTableSeqValue".toUpperCase() ) ) {
+				log.debug( " function 'getTableSeqValue' exists!" );
+				return;
+			}
+			InputStream is = OrientDBSchemaDefiner.class.getResourceAsStream( "getTableSeq.sql" );
+			Reader reader = new InputStreamReader( is, "utf-8" );
+			char[] chars = new char[2000];
+			CharBuffer buffer = CharBuffer.wrap( chars );
+			int realReadChars = reader.read( buffer );
+			log.debugf( "getTableSeqValue query: %s ; realReadChars: %d", new String( buffer.array() ).trim(), realReadChars );
+			connection.createStatement().execute( new String( buffer.array() ).trim() );
+		}
+		catch (SQLException | OException | IOException e) {
+			log.error( "Error!", e );
```

Comment:
No need to log the error, there is an exception already. Also `"Error!"` is too generic to be useful